tests/conftest.py

```diff
@@ -2739,21 +2750,66 @@ def taxprofiler_context(
     return cg_context
 
 
+@pytest.fixture(scope="function")
+def taxprofiler_deliverable_data(
+    taxprofiler_dir: Path, taxprofiler_case_id: str, sample_id: str
+) -> dict:
+    return {
+        "files": [
+            {
+                "path": f"{taxprofiler_dir}/{taxprofiler_case_id}/multiqc/multiqc_report.html",
+                "path_index": "",
+                "step": "report",
+                "tag": ["multiqc-html", "rna"],
+                "id": taxprofiler_case_id,
+                "format": "html",
+                "mandatory": True,
+            },
+        ]
+    }
+
+
+@pytest.fixture(scope="function")
+def taxprofiler_mock_deliverable(
+    taxprofiler_dir: Path, taxprofiler_deliverable_data: dict, taxprofiler_case_id: str
+) -> None:
```

Comment:
I'm not sure it's a good practice have fixtures returning None. Could we add this file context creation as part of the taxprofiler directory?

Reply:
Would that make sense to have?

```
@pytest.fixture(scope="function")
def taxprofiler_mock_deliverable(
    taxprofiler_dir: Path, taxprofiler_deliverable_data: dict, taxprofiler_case_id: str
):
    """Create taxprofiler deliverable file with dummy data and files to deliver."""
    Path.mkdir(
        Path(taxprofiler_dir, taxprofiler_case_id),
        parents=True,
        exist_ok=True,
    )
    Path.mkdir(
        Path(taxprofiler_dir, taxprofiler_case_id, "multiqc"),
        parents=True,
        exist_ok=True,
    )
    for report_entry in taxprofiler_deliverable_data["files"]:
        Path(report_entry["path"]).touch(exist_ok=True)
    WriteFile.write_file_from_content(
        content=taxprofiler_deliverable_data,
        file_format=FileFormat.JSON,
        file_path=Path(
            taxprofiler_dir, taxprofiler_case_id, taxprofiler_case_id + "_deliverables.yaml"
        ),
    )
    return taxprofiler_mock_deliverable
```

_Could we add this file context creation as part of the taxprofiler directory?_

Do you mean to create the taxprofiler_mock_deliverable file here?

```
@pytest.fixture(scope="function")
def taxprofiler_dir(tmpdir_factory, apps_dir: Path) -> Path:
    """Return the path to the Taxprofiler directory."""
    taxprofiler_dir = tmpdir_factory.mktemp("taxprofiler")
    return Path(taxprofiler_dir).absolute()

```

Reply:
Yes, let's make your `taxprofiler_mock_deliverable` fixture return the `taxprofiler_dir` and rename it to something like `taxprofiler_deliverables_dir` or `taxprofiler_filled_dir`

Reply:
Ok, I will do the same for rnafusion then since it is a parametrized test.

Reply:
Updated the tests.